lib/spack/spack/build_environment.py

```diff
@@ -321,12 +321,32 @@ def set_build_environment_variables(pkg, env, dirty=False):
         extra_rpaths = ':'.join(compiler.extra_rpaths)
         env.set('SPACK_COMPILER_EXTRA_RPATHS', extra_rpaths)
 
-    # Add bin directories from dependencies to the PATH for the build.
-    bin_dirs = reversed(filter(os.path.isdir, [
```

Comment:
It does for npm, but not for packages that depend on npm. The `bin` dirs are only added for explicit build dependencies. I don't think that package maintainers should have to specify that a package depending on `npm` also depends on `node-js`. That ought to come with using `npm` as a build dependency.